yarl/_helpers_c.pyx

```diff
@@ -0,0 +1,37 @@
+# cython: language_level=3
+
+cdef class cached_property:
```

Comment:
@bdraco how different is this from https://docs.python.org/3.8/library/functools.html#functools.cached_property? It exists in stdlib since Python 3.8.

Reply:
The functools version stores the cache in self.__dict__, and the yarl version stores in self._cache which means it also works with __slots__.  The cython implementation is faster than the pure python implementation in functools and better thought out and optimized for our use case.

Also the cached property implementation in functools had an undocumented threading lock which was removed in 3.12+

Reply:
Oh, I see. If this is copied across our projects, would it make sense to move it out into a separate lib?

Reply:
Yes it would make sense to break it out if we don't mind having another dep in the projects that use it.

Reply:
I don't mind personally. Especially, if it's not predicted to change often. Can't wait for us to be able to use ABI 3 (it's in the works in the new Cython) that would reduce the need for re-releases even more.

Reply:
I am working on the new package at https://github.com/bluetooth-devices/propcache reserved the name at https://pypi.org/project/propcache/

I will transfer it over when its ready

Reply:
- [ ] yarl https://github.com/aio-libs/yarl/pull/1169
- [ ] aiohttp https://github.com/aio-libs/aiohttp/pull/9394

Reply:
Honestly, I don't see a problem in hosting new projects in aio-libs right away.

Reply:
I was going to do that, but then I realized I didn’t have permissions to create it

Reply:
@bdraco if you go to the repo settings and click `Transfer`, does it let you select the org?

Reply:
Or I can just do it myself since you gave me the permissions...

Reply:
Done!

Reply:
Thank you!

Reply:
@bdraco I've improved your PyPI privileges in the org. You can try moving the project there too.

Reply:
@bdraco I'm going to file an issue in the repo with some things I'd like us to try out while the repo is fresh.

Reply:
https://github.com/aio-libs/propcache/issues/18

Reply:
I think it moved over ok but it presented a "You do not have permission to view this page" after I did it

Reply:
@bdraco that's caused by https://github.com/pypi/warehouse/issues/13558